src/NuGet.Core/NuGet.Protocol/HttpSource/HttpCacheUtility.cs

```diff
@@ -73,7 +73,11 @@ public static class HttpCacheUtility
                 FileShare.None,
                 BufferSize))
             {
+#if NETCOREAPP2_0_OR_GREATER
```

Comment:
In some other places you declared an extension method. 
Should we follow the same approach here?

Reply:
I tried doing the same thing, but I wasn't able to succeed. I started a thread in the Teams channel and Kartheek commented that maybe this is the best solution 

> ReadStreamAsync() don't have any overload method that accepts cancellation token in .NET Framework 472. We may have to live with If else suggestion you mentioned above. Even VS repo has If else thing for `ReadStreamAsync` [Utilities.cs - Repos (visualstudio.com)](https://devdiv.visualstudio.com/DevDiv/_git/VS?path=/src/env/shell/PackageFramework/Current/Shell/Utilities.cs&version=GBmain&_a=contents)

Reply:
I'm not sure I understand why this instance is different from other ones.

Reply:
[Stream.CopyToAsync()]( https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.copytoasync?view=netframework-4.7.2#system-io-stream-copytoasync(system-io-stream-system-int32-system-threading-cancellationtoken)) has an overloaded method that accepts `CancellationToken` in .NET Framework 472 whereas [HttpContent.ReadStreamAsync()](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpcontent.readasstreamasync?view=netframework-4.7.2) does not have any overloaded methods in .NET Framework 472. Hence, I suggested @martinrrm to add extension method for `Stream.CopyToAsync()` and use conditional compilation symbols for `HttpContent.ReadStreamAsync()`

Reply:
Unfortunately I don't follow.

Maybe a matrix would help: 

|  | net472 | netstandard2.0 | net5.0 |
|-|---------|------------------|---------|
| Stream.CopyToAsync | | | | 
| HttpContent.ReadStreamAsync | | | | 

You can always add more rows for methods.

Reply:
@martinrrm - Can you please add the details in the table so that it would be easy for review?

Reply:
Sorry for the delay in the response. I'm not sure what is expected in the matrix but this is the information I know.

|  | net472 | netstandard2.0 | net5.0 |
|-|---------|------------------|---------|
| Stream.CopyToAsync | No overload with `CancellationToken` only, but there is a overload with `CopyToAsync(destination, bufferSize, CancellationToken.None)` | No overload with `CancellationToken` only, but there is a overload with `CopyToAsync(destination, bufferSize, CancellationToken.None)` | Yes overload with `CancellationToken` | 
| HttpContent.ReadStreamAsync | No overload with `CancellationToken` | No overload with `CancellationToken` | Yes overload with `CancellationToken` | 

After doing this matrix and with the comments from the above discussion I think is better to do a conditional compile and avoid adding new API's.

Reply:
Looks like you're using conditional compilation everywhere. 

I'm fine with that as it's consistent. 

My feedback was primarily that we either use a conditional compilation at the site of the offending code or we declare an extension method that's conditionally compiled.